cmd/hzn/main.go

```diff
@@ -485,13 +486,54 @@ func (h *hubRunner) Run(args []string) int {
 
 	deployment := os.Getenv("K8_DEPLOYMENT")
 
+	// We want to have the control client filter use ConsulHealth,
+	// so we establish that here for use as a filter.
+
+	ccfg := consul.DefaultConfigWithLogger(L)
+
+	// Check that consul is available first.
+
+	cc, err := consul.NewClient(ccfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var ch *hub.ConsulHealth
+
+	instanceId := pb.NewULID()
+
+	var filter func(serv *pb.ServiceRoute) bool
+
+	status := cc.Status()
+	leader, err := status.Leader()
+	if err == nil {
+		L.Info("consul running, leader detected", "leader", leader)
+		L.Info("starting consul health monitoring")
+
+		ch, err = hub.NewConsulHealth(instanceId.SpecString(), ccfg)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		filter = func(serv *pb.ServiceRoute) bool {
+			return ch.Available(serv.Hub.SpecString())
+		}
+	} else {
+		L.Warn("consul not available, no consul health monitoring done")
+		filter = func(serv *pb.ServiceRoute) bool {
+			return true
```

Comment:
Why do we default to `true` if consul is not available? The connection between this and the comment in client.go is not immediately obvious to me (`// If the user has requested the ability to filter the services also then give them the chance here.`)

Reply:
This is effectively disabling this filter function, allowing it to use the default filter logic only. Effectively, this filter gives us a chance to shoehorn in a consul availability check into the lookup logic, but if consul isn't available, we just leave the lookup logic running as it normally would.